appendices.tex

```diff
@@ -370,6 +370,152 @@ \chapter{Listing of application roles}
+\end{description}
+
+\subsubsection{'programs' section}
+\label{subsubsec-global-config-ledgersmb-conf-programs}
+
+\begin{description}[style=nextline]
+	\item [gzip] @@@TODO
+\end{description}
+
+
+\subsubsection{'mail' section}
+\label{subsubsec-global-config-ledgersmb-conf-mail}
+
+\begin{description}[style=nextline]
+	\item [sendmail] @@@TODO
+	\item [smtphost] @@@TODO
+	\item [smtptimeout] @@@TODO
+	\item [backup\_email\_from] @@@TODO
+\end{description}
+
+\subsubsection{'printers' section}
+\label{subsubsec-global-config-ledgersmb-conf-printers}
+
+This section contains a list of printers and the commands
+to be executed in order for the output to be sent to the given printer with
+the document to be printed fed to the program through standard input.
+The example below shows how to send standard input data to a printer called ``laser''
+when selecting the item ``Laser'' in the LedgerSMB printer selection drop down.
+
+\texttt{
+	Laser    = lpr -Plaser
+}
+
+
+\subsubsection{'database' section}
+\label{subsubsec-global-config-ledgersmb-conf-database}
+
+\begin{description}[style=nextline]
+	\item [host] Name of the host to connect to. See the documentation of the {\tt -h} command line option at 
+	\url{http://www.postgresql.org/docs/9.2/static/app-psql.html}
+	for more information (documentation unchanged since before 8.3, so applies to older versions as well)
+	\item [port] Port number to be used to connect to. See the documentation of the {\tt -p} option at the
+	same URL as the previous item
+	\item [default\_db] Database to connect to when the ``Company'' entry field in the login screen is left blank
+	\item [db\_namespace] The name space the company resides in; expert setting -- should not be used (default is
+	``public'')
+	\item [contrib\_dir] Path of the directory in which the PostgreSQL contrib modules (extensions) are
+	located; this setting is used by setup.pl when creating new companies or rebuilding (upgrading)
+	existing ones
+	\item [sslmode] Selects whether to use \gls{SSL} over TCP connections or not; can be ``require'', ``allow'',
+	``prefer'' or ``disable''
+\end{description}
+
+\subsubsection{log4perl\_config\_modules\_loglevel}
+\label{subsubsec-global-config-ledgersmb-conf-log4perl}
+
+@@@ This is a list of \textless module.name \textgreater = \textless level \textgreater
+
+This is an advanced setting not typically touched.
+
+
+\subsection{pos.conf.pl}
+\label{subsec-global-config-pos-conf}
+
+
+@@@ no idea what should go in here. to be investigated.
+
+\subsection{templates}
+\label{subsec-global-config-templates}
+
+LedgerSMB uses templates mainly for two things:
+
+\begin{itemize}
+	\item Output documents such as pick lists and invoices
+	\item The HTML User Interface
+\end{itemize}
+
+This setting relates to the former, which are meant to be copied and customized by users. The system
+supports having multiple sets in a global setup, where companies can be configured to use different
+sets.

```

Comment:
Right. Even more specifically: in the version this document describes, the role of this value is to help "setup.pl" find the available options which are copied into the database at creation time. Once copied into the database, the on-disk templates aren't used anymore (with minor exceptions).

Reply:
Which minor exceptions?

Reply:
A template may include a nested template from `templates/lib/`. This happens for "table output" (`dynatable.*`) output and for the `preamble.tex` which sets the latex/xelatex font, etc.

Reply:
BTW, it's actually the intent that even those use-cases be eliminated, but loading dynatable into the database as an *editable* template didn't seem like the right thing to do...

Reply:
This was meant to be in the deprecated ledgersmb.conf sections above it.  Since it no longer applies I remove it.